java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java

```diff
@@ -0,0 +1,403 @@
+  @Override
+  public int getBufferSizeFor(int valueCount) {
+    if (valueCount == 0) {
+      return 0;
+    }
+
+    int innerVectorValueCount = 0;
+
+    for (int i = 0; i < valueCount; i++) {
+      innerVectorValueCount += sizeBuffer.getInt(i * SIZE_WIDTH);
+    }
+
+    return (valueCount * OFFSET_WIDTH) + (valueCount * SIZE_WIDTH) +
+        vector.getBufferSizeFor(innerVectorValueCount);
+  }
+
+  @Override
+  public Iterator<ValueVector> iterator() {
+    return Collections.<ValueVector>singleton(getDataVector()).iterator();
+  }
+
+  @Override
+  public void clear() {
+    offsetBuffer = releaseBuffer(offsetBuffer);
+    sizeBuffer = releaseBuffer(sizeBuffer);
+    vector.clear();
+    valueCount = 0;
+    super.clear();
+  }
+
+  @Override
+  public void reset() {
+    offsetBuffer.setZero(0, offsetBuffer.capacity());
+    sizeBuffer.setZero(0, sizeBuffer.capacity());
+    vector.reset();
+    valueCount = 0;
+  }
+
+  @Override
+  public ArrowBuf[] getBuffers(boolean clear) {
+    return new ArrowBuf[0];
+  }
+
+  @Override
+  public int getValueCount() {
+    return valueCount;
+  }
+
+  @Override
+  public boolean isNull(int index) {
+    return false;
+  }
+
+  @Override
+  public void setValueCount(int valueCount) {
+    this.valueCount = valueCount;
+    while (valueCount > getOffsetBufferValueCapacity()) {
+      reallocateBuffers();
+    }
+    final int childValueCount = valueCount == 0 ? 0 : getLengthOfChildVector();
+    vector.setValueCount(childValueCount);
+  }
+
+  private int getLengthOfChildVector() {
+    int length = 0;
+    for (int i = 0; i <= valueCount; i++) {
+      length += sizeBuffer.getInt(i * SIZE_WIDTH);
+    }
+    return length;
+  }
+
+  /**
+   * Initialize the data vector (and execute callback) if it hasn't already been done,
+   * returns the data vector.
+   */
+  public <T extends ValueVector> AddOrGetResult<T> addOrGetVector(FieldType fieldType) {
+    boolean created = false;
+    if (vector instanceof NullVector) {
+      vector = fieldType.createNewSingleVector(defaultDataVectorName, allocator, repeatedCallBack);
+      // returned vector must have the same field
+      created = true;
+      if (repeatedCallBack != null &&
+              // not a schema change if changing from ZeroVector to ZeroVector
+              (fieldType.getType().getTypeID() != ArrowType.ArrowTypeID.Null)) {
+        repeatedCallBack.doWork();
+      }
+    }
+
+    if (vector.getField().getType().getTypeID() != fieldType.getType().getTypeID()) {
+      final String msg = String.format("Inner vector type mismatch. Requested type: [%s], actual type: [%s]",
+              fieldType.getType().getTypeID(), vector.getField().getType().getTypeID());
+      throw new SchemaChangeRuntimeException(msg);
+    }
+
+    return new AddOrGetResult<>((T) vector, created);
+  }
+
+  protected void replaceDataVector(FieldVector v) {
+    vector.clear();
+    vector = v;
+  }
+
+  public boolean isEmpty(int index) {
+    return false;
```

Comment:
This seems wrong?

Reply:
I wasn't confident with this either. But I referred to the `ListVector` base class, and it has the following (maybe not related to this), any thoughts? 

https://github.com/apache/arrow/blob/53859262ea988f31ce33a469305251064b5a53b8/java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java#L364

Reply:
Throw or make abstract

Reply:
Meaning should we do the same for ListVector (I mean not in this PR, in another one?)

Reply:
Ideally yes

Reply:
Issue created: https://github.com/apache/arrow/issues/41590